### PR TITLE
✅ Improving CI tests: web/server/tests/integration/02/test_rabbit.py

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

✅ This PR aims to fix the flaky ``services/web/server/tests/integration/02/test_rabbit.py`` test-suite and improve the CI stability.

- ♻️ split single ``test_rabbit_websocket_computation `` into smaller independent ``tests_`` cases
- ♻️ cleaned up pytest-simcore fixtures used in this test-suite
- ♻️ cleaned up some of the web-server's modules used in this test

## Related issue/s

Reviewing flaky integration tests in CI

## How to test

```
make devenv
source .venv/bin/activate
cd services/web/server
make install-dev
pytest -vv tests/integration/02/test_rabbit.py
```

### CODE REVIEW

**Review by GitHK — APPROVED** [review]
👍

File: packages/pytest-simcore/src/pytest_simcore/rabbit_service.py [hunk 1]
```diff
@@ -18,11 +18,27 @@
...
+    connection = await aio_pika.connect(url)
+    await connection.close()
+
+
+# FIXTURES ------------------------------------------------------------------------------------
+
+
 @pytest.fixture(scope="module")
-def loop(request) -> asyncio.AbstractEventLoop:
+def loop(request) -> Iterator[asyncio.AbstractEventLoop]:
```
Comment (sanderegg) [inline packages/pytest-simcore/src/pytest_simcore/rabbit_service.py:41]: I am actually confused whether the return type should be one of Iterable, Iterator, Generator....
see this [https://github.com/pytest-dev/pytest/discussions/7809](https://github.com/pytest-dev/pytest/discussions/7809)

File: services/web/server/src/simcore_service_webserver/socketio/module_setup.py [hunk 2]
```diff
@@ -5,25 +5,30 @@
...
     # ----------------------------------------------
     # TODO: temporary, just to check compatibility between
     # trafaret and pydantic schemas
     assert_valid_config(app)
     # ---------------------------------------------
-    mgr = None
-    sio = AsyncServer(async_mode="aiohttp", client_manager=mgr, logging=log)
+
+    # SEE https://github.com/miguelgrinberg/python-socketio/blob/v4.6.1/docs/server.rst#aiohttp
+    # TODO: ujson to speed up?
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/socketio/module_setup.py:28]: or orjson... seems to be the best of the best...

**Review by sanderegg — APPROVED** [review]
Very nice! let's hope the garbage collector trick makes it solid like a rock!

File: packages/pytest-simcore/src/pytest_simcore/rabbit_service.py (hunk 1 above)
Comment (pcrespov) [reply packages/pytest-simcore/src/pytest_simcore/rabbit_service.py:41]: ``Iterable`` or ``Iterator`` are specific cases of ``Generator[YieldType, SendType, ReturnType]`` but that does not have send or return  , i.e. ``Iterator = Generator[YieldType, None, None]``.  

See more details and examples in https://docs.python.org/3/library/typing.html#typing.Generator

Functionalitywise, ``Iterable`` and ``Iterator`` are equivalent and the difference is a bit more subtle. An iterator is iterable but not every iterable is an iterator (there are othr ways for python to deduce how to iterate). Taking from [this reference ](https://www.geeksforgeeks.org/python-difference-iterable-iterator):

> Iterable is an object, which one can iterate over. It generates an Iterator when passed to iter() method. Iterator is an > object, which is used to iterate over an iterable object using __next__() method. Iterators have __next__() method, > which returns the next item of the object.

**Pull request merged**